### PR TITLE
fix(tui): source new-session repo picker from scanner cache

### PR DESCRIPTION
**Pull request by deepaks7n**
## Problem

The 2-screen new-session redesign (PR #166, `48c1ea9`) builds the repo picker's local list from `state.workspaces`, which only holds repos that have an **active session**. With no live session the fuzzy filter has nothing to match, so local repos are unfindable when creating a new session — and the scanned repo list in `~/.agents-in-a-box/cache/repositories.json` is never consulted.

## Fix

- **Source the picker from the `WorkspaceScanner` cache** (`RepositoryCache::load()`) — a cheap JSON read, not a synchronous filesystem walk, so it avoids the event-loop freeze that motivated dropping the inline scan in the redesign.
- **Filter to directories that still exist**, so a repo deleted or moved since the last scan can't appear as a selectable dead row.
- **Refresh the cache on a background `spawn_blocking` task** (read-through `scan()`), so a newly-created repo surfaces on a later open. Falls back to active-session workspaces only when no cache exists yet (first run).
- **Make `RepositoryCache::save()` atomic** (temp file + rename via `tempfile::NamedTempFile`), so a reader or a second concurrent rescan never observes a half-written cache.

A pure helper (`picker_local_paths`) is extracted so the resolution logic is unit-tested.

## Tests

- New unit tests: existence-filter and cache-absent fallback.
- `pick_repo` (5) and `workspace_scanner` (6) suites pass; release build is clean and adds no new warnings.

## Release Notes

* **Bug Fixes**
  * Improved session picker path resolution for cached repositories
  * Enhanced cache file writing reliability
  * Background workspace scanning no longer blocks the interface

* **Tests**
  * Added tests for cache-aware path resolution

[Review Change Stack](https://app.coderabbit.ai/change-stack/stevengonsalvez/agents-in-a-box/pull/170?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: ainb-tui/crates/ainb-core/src/app/events.rs [hunk 1]
```diff
@@ -406,6 +406,74 @@ fn derive_repo_label(source: &crate::git::repo_source::RepoSource) -> String {
...
+    match cache {
+        Some(c) => c
+            .repositories
+            .into_iter()
+            .filter(|r| r.path.is_dir())
+            .map(|r| r.path)
+            .collect(),
+        None => workspaces.iter().map(|w| w.path.clone()).collect(),
+    }
+}
```
[the remark this answers is not in the document]
Comment (deepaks7n) [reply ainb-tui/crates/ainb-core/src/app/events.rs:438]: Keeping the `is_dir()` filter here, deliberately.

It's O(N) lightweight `stat`s over a list of *already-known* paths — not the recursive `WorkspaceScanner::scan()` (readdir + `git2` open per directory) that motivated moving scanning off the UI thread. For a few hundred local repos it's sub-millisecond to low-single-digit ms on open, a different order of magnitude from the scan.

Dropping it means a repo deleted or moved since the last cache write shows up as a selectable row and then fails later at worktree creation in a missing directory — a more confusing outcome than the row simply not appearing. The background rescan doesn't help the *current* open (it refreshes the cache for the next one), so the filter is what keeps this open honest.

If UI-thread I/O at very large repo counts (thousands, or network-mounted scan roots) is the real concern, the cleaner fix is to validate existence at *selection* time rather than dropping the guard — happy to add that if you'd prefer.

File: ainb-tui/crates/ainb-core/src/app/events.rs [hunk 2]
```diff
@@ -406,6 +406,74 @@ fn derive_repo_label(source: &crate::git::repo_source::RepoSource) -> String {
...
+    }
+
+    #[test]
+    fn keeps_only_cache_entries_whose_dir_exists() {
+        let dir = tempfile::TempDir::new().unwrap();
+        let real = dir.path().to_path_buf();
+        let gone = real.join("gone");
+        let out = picker_local_paths(Some(cache_with(vec![real.clone(), gone])), &[]);
+        assert_eq!(out, vec![real]);
+    }
```
[the remark this answers is not in the document]
Comment (deepaks7n) [reply ainb-tui/crates/ainb-core/src/app/events.rs:476]: Leaving this test as-is: it's contingent on removing the `is_dir()` filter above, which I'm keeping (see the reply there). The test intentionally pins that a path which no longer exists is dropped from the picker. I did add a sibling test (`falls_back_to_workspace_paths_when_cache_filters_to_empty`) for the empty-filtered-cache → workspace fallback per CodeRabbit's note.

File: ainb-tui/crates/ainb-core/src/app/events.rs [hunk 3]
```diff
@@ -406,6 +406,74 @@ fn derive_repo_label(source: &crate::git::repo_source::RepoSource) -> String {
...
+) -> Vec<std::path::PathBuf> {
+    match cache {
+        Some(c) => c
+            .repositories
+            .into_iter()
+            .filter(|r| r.path.is_dir())
+            .map(|r| r.path)
+            .collect(),
+        None => workspaces.iter().map(|w| w.path.clone()).collect(),
+    }
```
[the remark this answers is not in the document]
Comment (deepaks7n) [reply ainb-tui/crates/ainb-core/src/app/events.rs:437]: Good catch — fixed in af33323. An empty filtered cache (no cache file yet, or every cached repo deleted/moved) now falls back to `state.workspaces` the same as a cache miss, with a `falls_back_to_workspace_paths_when_cache_filters_to_empty` unit test.
